### PR TITLE
[Sparse] Only support scalar or vector values in SparseMatrix creators

### PR DESCRIPTION
**Pull request by czkkkkkk**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by jermainewang — APPROVED** [review]

File: python/dgl/sparse/sparse_matrix.py [hunk 1]
```diff
@@ -565,6 +568,10 @@ def from_csr(
     if val is None:
         val = torch.ones(indices.shape[0]).to(indptr.device)
 
+    assert (
```
Comment (frozenbugs) [inline python/dgl/sparse/sparse_matrix.py:571]: Move assert above the first if check.

File: python/dgl/sparse/sparse_matrix.py [hunk 2]
```diff
@@ -482,6 +482,9 @@ def from_coo(
         shape = (torch.max(row).item() + 1, torch.max(col).item() + 1)
     if val is None:
         val = torch.ones(row.shape[0]).to(row.device)
+    assert (
```
Comment (frozenbugs) [inline python/dgl/sparse/sparse_matrix.py:485]: Move assert above the first if check.

File: python/dgl/sparse/sparse_matrix.py [hunk 3]
```diff
@@ -649,6 +656,10 @@ def from_csc(
     if val is None:
         val = torch.ones(indices.shape[0]).to(indptr.device)
 
+    assert (
```
Comment (frozenbugs) [inline python/dgl/sparse/sparse_matrix.py:659]: ditto

File: python/dgl/sparse/sparse_matrix.py [hunk 4]
```diff
@@ -686,6 +697,11 @@ def val_like(mat: SparseMatrix, val: torch.Tensor) -> SparseMatrix:
     values=tensor([2, 2, 2]),
     shape=(3, 5), nnz=3)
     """
+
```
Comment (frozenbugs) [inline python/dgl/sparse/sparse_matrix.py:700]: redundant blank line.

**Comment by frozenbugs** [comment]
Also update diag matrix?

File: python/dgl/sparse/sparse_matrix.py (hunk 1 above)
Comment (czkkkkkk) [reply python/dgl/sparse/sparse_matrix.py:571]: This check should be after `if val is None`.

**Comment by czkkkkkk** [comment]
> Also update diag matrix?

Added.

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**